packages/strapi/lib/services/core-store.js

```diff
@@ -105,9 +105,28 @@ const createCoreStore = ({ environment: defaultEnv, db }) => {
       }
     }
 
+    async function deleteFn(params = {}) {
+      const { key, environment = defaultEnv, type, name, tag = '' } = Object.assign(
+        {},
+        source,
+        params
+      );
+      const prefix = `${type}${name ? `_${name}` : ''}`;
+      const where = {
+        key: `${prefix}_${key}`,
+        environment,
+        tag,
+      };
+      const data = await db.query('core_store').findOne(where);
```

Comment:
I think you can do q delete directly with your where close actually. delete({ key: 'abcd',...})

Reply:
It is done! I did the check because I wasn't sure all different DB's accept a delete on a non-existing row.